File: backend/app/schemas/season.py
from datetime import date, datetime

from pydantic import BaseModel


class SeasonCreate(BaseModel):
    name: str
    start_date: date
    end_date: date


class SeasonUpdate(BaseModel):
    name: str | None = None
    start_date: date | None = None
    end_date: date | None = None


class SeasonResponse(BaseModel):
    id: int
    name: str
    start_date: date
    end_date: date
    status: str
    created_at: datetime | None = None

File: frontend/shared/types/season.ts
import type { SeasonStatus } from './enums'

export interface Season {
  id: number
  name: string
  start_date: string
  end_date: string
  status: SeasonStatus
  created_at: string
  updated_at: string
}

export interface SeasonCreate {
  name: string
  start_date: string
  end_date: string
}

export interface SeasonUpdate {
  name?: string
  start_date?: string
  end_date?: string
}
